lib/src/layer/tile_layer/tile_layer.dart

```diff
@@ -602,6 +644,9 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
       onLoadError: _onTileLoadError,
       onLoadComplete: (coordinates) {
         if (pruneAfterLoad) _pruneIfAllTilesLoaded(coordinates);
+        setState(() {
+          //Refresh the widget to display the loaded tile
+        });
```

Comment:
This feels like a workaround; maybe it isn't though? Why is it required now and not before?

Reply:
Previously each tile rebuilt it's own state when image was loaded. Needed a place to call` setState()` to refresh TilePainter. 
Seems like `onLoadComplete` is appropriate place to call that.

Reply:
Now looking deeper into this I noticed that fadeIn animation is running, but not animating opacity.  Perhaps can rebuild state within animation controller listener and set opacity to image in TilePainter?

Reply:
Opacity is updating, but if there is no rebuild it gets stuck on that opacity level. 
https://github.com/fleaflet/flutter_map/assets/69913791/f12260bf-bd5a-4e4e-9849-dd865896f98b

Reply:
I think an `if (mounted)` should be added in case the onLoadComplete is called after the widget has been destroyed.